src/modules/commander/commander_params.c

```diff
@@ -769,3 +769,66 @@ PARAM_DEFINE_INT32(COM_FLIGHT_UUID, 0);
  * @group Mission
  */
 PARAM_DEFINE_INT32(COM_TAKEOFF_ACT, 0);
+
+/**
+ * Airspeed Consistency Threshold.
+ *
+ * This scales the minimum airspeed inconsistency required to trigger a failsafe. Increase to make the check less sensitive, decrease to make it more sensitive. The failsafe response is controlled by the COM_ASPD_FS_ACT parameter.
+ *
+ * @min 0.5
+ * @max 2.0
+ * @group Commander
+ */
+PARAM_DEFINE_FLOAT(COM_TAS_FS_INNOV, 1.0f);
```

Comment:
Based on flight testing and review of flight data, we feel like 1.0 causes too many false positives for COM_TAS_INNOV.  It is one thing to throw a flag indicating airspeed readings aren't in agreement with expected values and it is another to declare them bad enough to trigger a failsafe.  Hence, I suggested the default for COM_TAS_FS_INNOV = 2.0.  Thoughts?

Reply:
That would be acceptable because we still have the load factor test. What value did the airspeed innovation ratio peak at for the false positives you experienced?

Reply:
Sometimes the airspeed innovations can get pretty large, especially in windy conditions with a lot of turning.  Here are two flight logs with a lot of false positives.  Airspeed was generally working, might be not calibrated perfectly, but it was reasonable.  The biggest contributor here is high winds.  You will see that we were changing params mid-flight to test the sensitivity to false positives:
https://review.px4.io/plot_app?log=d9a42e33-32b2-4530-ad71-bc9d917465f6
![image](https://user-images.githubusercontent.com/12117027/55690676-4caf5b00-595a-11e9-88be-82401025f017.png)

https://review.px4.io/plot_app?log=fa18a869-1a7b-4a5f-88ef-8dd5e6c22e79
![image](https://user-images.githubusercontent.com/12117027/55690683-5d5fd100-595a-11e9-8dc7-c141ed31b182.png)

Reply:
@Antiheavy  Thanks for sharing those.

Besides increasing the threshold we can also reduce false positives by integrating the error when above the threshold so that larger transient conditions do not trigger, but a smaller persistent condition does.

I will have a go at this and also look at the log where the wind vector initialised incorrectly.

Reply:
> Besides increasing the threshold we can also reduce false positives by integrating the error when above the threshold so that larger transient conditions do not trigger, but a smaller persistent condition does.

@priseborough this is a neat idea, but maybe it isn't worth the effort.  It could be worth it if the solution allowed us to eliminate the need to parameterize COM_TAS_FS_INNOV, making this feature simpler for users/integrators.  However, if we need a parameter to tune COM_TAS_FS_INNOV for different types of vehicles and users, then we might as well keep it a bit simpler to understand as a simple scaler on the airspeed test ratio.  I feel with COM_ASPD_FS_DLY the solution is equally "safe" even with some false positives.

Reply:
In theory with a sensible integral error threshold we should be able to have the same COM_TAS_FS_INNOV for different vehicles because that is based on the innovation test ratio which if above 1 indicates the airspeed has been rejected by the EKF. 

However in practice there will be many users that are flying with badly calibrated airspeed installations that are likely to fail the test. For example see the ones supplied by a FW tester for the 1.9 release here: https://github.com/PX4/Firmware/issues/11785 where there was no awareness of the large impact of static source position error. That log would have triggered any sensible threshold, eg:

![Screen Shot 2019-04-22 at 9 32 47 am](https://user-images.githubusercontent.com/3596952/56476879-176e4580-64e2-11e9-9282-058ec8fda71d.png)

However in this example I think the user should have been alerted that they have a bad calibration. One way around this is for the initial implementation would be to use the innovation check in a 'warning only' capacity and relay on the load factor check for the failsafe.

I will implement the 'integral' check method and provide a link to it for further discussion here: https://github.com/PX4/Firmware/pull/11846